store/pruning/manager_test.go

```diff
@@ -0,0 +1,84 @@
+package pruning
+
+import (
+	"testing"
+
+	dbm "github.com/cosmos/cosmos-db"
+	"github.com/stretchr/testify/suite"
+
+	"cosmossdk.io/log"
+	"cosmossdk.io/store/v2"
+	"cosmossdk.io/store/v2/commitment/iavl"
+	"cosmossdk.io/store/v2/storage/sqlite"
+)
+
+type PruningTestSuite struct {
+	suite.Suite
+
+	manager *Manager
+	ss      store.VersionedDatabase
+	sc      store.Committer
+}
+
+func TestPruningTestSuite(t *testing.T) {
+	suite.Run(t, &PruningTestSuite{})
+}
+
+func (s *PruningTestSuite) SetupTest() {
+	noopLog := log.NewNopLogger()
+
+	ss, err := sqlite.New(s.T().TempDir())
+	s.Require().NoError(err)
+
+	sc := iavl.NewIavlTree(dbm.NewMemDB(), noopLog, iavl.DefaultConfig())
+
+	s.manager = NewManager(ss, sc, noopLog)
+	s.ss = ss
+	s.sc = sc
+}
+
+func (s *PruningTestSuite) TearDownTest() {
+	s.manager.Start()
+	s.manager.Stop()
+}
+
+func (s *PruningTestSuite) TestPruning() {
+	s.manager.SetCommitmentOptions(Option{4, 2})
+	s.manager.SetStoreOptions(Option{4, 3})
+	s.manager.Start()
+
+	// write 10 batches
+	for i := 0; i < 10; i++ {
+		cs := store.NewChangeset()
+		cs.Add([]byte("key"), []byte("value"))
```

Comment:
Could we make sure that each iteration of i has a key and value that have the same suffix? So "key1", "value1" etc this will help with much clearer testing and verifying that version 5 returns the right results after pruning. If the implementation at all were to erroneously return the wrong result, we'd never catch that problem if you don't have the right ordinal numbers to verify against.